build/generated.bzl

```diff
@@ -0,0 +1,149 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The go_prefix for this repository, needed to map relative package paths into
+# a full-resolved go package.
+go_prefix = "k8s.io/kubernetes"
+
+# Pretty-prints the provided dict -> dict -> dict as a dict -> dict -> list,
+# appending lines into output.
+def _format_output_dict(id, tags, output):
+    output.append("%s = {" % id)
+
+    for tag in tags.keys():
+        d = tags[tag]
+        output.append("%s%s: {" % (4 * " ", repr(tag)))
+        for k in d.keys():
+            output.append("%s%s: [" % (8 * " ", repr(k)))
+            for v in d[k].keys():
+                output.append("%s%s," % (12 * " ", repr(v)))
+            output.append("%s]," % (8 * " "))
+
+        output.append("%s}," % (4 * " "))
+
+    output.append("}\n")
+
+def _find_generator_tag_pkgs_impl(repo_ctx):
+    repo_root = repo_ctx.path(repo_ctx.attr._workspace).dirname
+
+    result = repo_ctx.execute(
+        ["grep", "--color=never", "-Hrox", "--include=*.go", "\\s*//\s*+k8s:\\S*=\\S*\\s*", repo_root],
+        quiet = True,
+    )
+    if result.return_code:
+        fail("failed searching for generator build tags: %s" % result.stderr)
+
+    # Maps tag names -> go packages -> values found for that tag in that package
+    tags_pkgs_values = {}
+
+    # Maps tag names -> values found for that tag -> go packages with those tag/value mappings
+    tags_values_pkgs = {}
+
+    for line in result.stdout.splitlines():
+        # Each line looks something like
+        # /a/full/path/to/file.go:+k8s:foo=bar
+
+        # First split out the filename from the matched tag blog
+        fname, _, match = line.partition(":")
```

Comment:
Will windows `c:\\whatever` paths pose a problem? Do we care?

Reply:
`ctx.configuration.host_path_separator`?
https://docs.bazel.build/versions/master/skylark/lib/configuration.html#host_path_separator

This seems pretty easy to fix, if we care..